integration-tests/jiva/node-stickiness/replicadeletion/jiva_replica_deletion_test.go

```diff
@@ -0,0 +1,164 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package replicadeletion
+
+import (
+	"fmt"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	clientpvc "github.com/openebs/maya/pkg/kubernetes/persistentvolumeclaim/v1alpha1"
+	pod "github.com/openebs/maya/pkg/kubernetes/pod/v1alpha1"
+	clientsc "github.com/openebs/maya/pkg/kubernetes/storageclass/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	k8serror "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	// auth plugins
+	_ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
+)
+
+const (
+	testTimes = 20
+)
+
+var _ = Describe("[jiva] [node-stickiness] jiva replica pod node-stickiness test", func() {
+	var (
+		// replicaLabel consist of defaultReplicaLabel and coressponding
+		// pvcLabel
+		replicaLabel string
+		// ctrlLabel consist of defaultReplicaLabel and coressponding
+		// pvcLabel
+		ctrlLabel string
+		//podListObj holds the PodList instance
+		podListObj    *corev1.PodList
+		podKubeClient *pod.KubeClient
+		// defaultReplicaLabel represents the jiva replica
+		defaultReplicaLabel = "openebs.io/replica=jiva-replica"
+		// defaultCtrlLabel represents the jiva controller
+		defaultCtrlLabel = "openebs.io/controller=jiva-controller"
+		// defaultPVCLabel represents the default OpenEBS PVC label key
+		defaultPVCLabel       = "openebs.io/persistent-volume-claim="
+		storageEngine         = "jiva"
+		replicaCount          = "1"
+		openebsCASConfigValue = "- name: ReplicaCount\n  Value: " + replicaCount
+		scName                = "jiva-single-replica"
+		pvcName               = "jiva-vol1-1r-claim"
+		testNamespace         = "jiva-rep-delete-ns"
+		accessModes           = []corev1.PersistentVolumeAccessMode{corev1.ReadWriteOnce}
+		capacity              = "5G"
+		//TODO: following variables should be moved in framework or openebs-artifacts
+		openebsCASType     = "cas.openebs.io/cas-type"
+		openebsCASConfig   = "cas.openebs.io/config"
+		openebsProvisioner = "openebs.io/provisioner-iscsi"
+	)
+	BeforeEach(func() {
+		var err error
+		By("Build jiva-single-replica storageclass and deploy it")
+		annotations := map[string]string{
+			openebsCASType:   storageEngine,
+			openebsCASConfig: openebsCASConfigValue,
+		}
+		buildSCObj, err := clientsc.NewBuilder().
+			WithName(scName).
+			WithAnnotations(annotations).
+			WithProvisioner(openebsProvisioner).Build()
+		Expect(err).ShouldNot(HaveOccurred())
+
+		_, err = clientsc.NewKubeClient(clientsc.WithKubeConfigPath(kubeConfigPath)).Create(buildSCObj)
+		Expect(err).ShouldNot(HaveOccurred())
+
+		By("Build and deploy PVC using jiva-single-replica storageClass in jiva-rep-delete-ns namespace")
```

Comment:
Split build from deploy in By() function

Reply:
Done